ui/src/dashboards/components/CellEditorOverlay.js

```diff
@@ -87,31 +124,44 @@ class CellEditorOverlay extends Component {
   }
 
   handleSaveCell() {
-    const {queriesWorkingDraft, cellWorkingType, cellWorkingName} = this.state
+    const {
+      queriesWorkingDraft,
+      cellWorkingType: type,
+      cellWorkingName: name,
+      axes,
+    } = this.state
+
     const {cell} = this.props
 
-    const newCell = _.cloneDeep(cell)
-    newCell.name = cellWorkingName
-    newCell.type = cellWorkingType
-    newCell.queries = queriesWorkingDraft.map(q => {
+    const queries = queriesWorkingDraft.map(q => {
       const timeRange = q.range || {upper: null, lower: ':dashboardTime:'}
       const query = q.rawText || buildInfluxQLQuery(timeRange, q)
-      const label = q.rawText ? '' : `${q.measurement}.${q.fields[0].field}`
 
       return {
         queryConfig: q,
         query,
-        label,
       }
     })
 
-    this.props.onSave(newCell)
+    this.props.onSave({
+      ...cell,
+      name,
+      type,
+      queries,
+      axes,
+    })
   }
 
   handleSelectGraphType(graphType) {
     this.setState({cellWorkingType: graphType})
   }
 
+  handleSelectDisplayOptions(isDisplayOptionsTabOpen) {
+    return () => {
```

Comment:
not sure you need this outer `return` -- i _think_ you can just call `this.setState` directly inside this function, since i believe `handleSelectDisplayOptions` itself is the function wrapper i think you're looking for 🤖  EDIT: i see that this is necessary given how it's being called further down. personally i'd prefer the anonymous function to be placed in the onClick handler, but that's just a style thing.